pkg/executor/statefulset.go

```diff
@@ -38,10 +38,28 @@ import (
 // that will always be placed on shared persistent storage (when available).
 var defaultMountFolders = []string{"/etc"}
 
-// appConfigDefaultMountFolders identifies set of member filesystems directories
-// that will always be placed on shared persistent storage, if app config is provided
-// for a role
-var appConfigDefaultMountFolders = []string{"/etc", "/opt", "/usr"}
+// appConfigDefaultMountFolders identifies set of member filesystems
+// directories that will always be placed on shared persistent storage, if app
+// config is provided for a role, and if that role's package has
+// useNewSetupLayout set true. Note that the code in getStatefulset assumes
+// this is a superset of the defaultMountFolders list.
+var appConfigDefaultMountFolders = []string{
+	"/etc",
```

Comment:
why not /etc/guestconfig

Reply:
So, the underlying concern is that any service config changes that the app setup does need to be persisted.

Perhaps ideally this would be part of the app's persistdirs. But no app currently specifies this (because they are -- not unreasonably in this case -- depending on the fact that KD captures it) and if we dropped it from the KD defaults then almost every app would need to specify it. I dunno. I decided just to leave it.